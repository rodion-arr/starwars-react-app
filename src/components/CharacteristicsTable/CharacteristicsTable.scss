.detail-spec-table {
  border-color: #373b3e;
  display: flex;
  justify-content: center;

  & > :not(caption) > * > * {
    padding: .5rem .5rem;
    background-color: #1e1d1d;
    border-bottom-width: 1px;
    box-shadow: inset 0 0 0 9999px transparent;
    text-transform: capitalize;
  }

  &__table {
    width: 500px;
  }

  * {
    color: inherit;
  }
}
