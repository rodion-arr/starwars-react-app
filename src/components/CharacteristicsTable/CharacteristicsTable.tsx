import React from "react";
import './CharacteristicsTable.scss';

interface Props {
  className?: string;
  rows: Record<string, string | JSX.Element>;
}

export const CharacteristicsTable = ({
  className = '',
  rows,
}: Props): JSX.Element | null => {
  const charProperties = Object.keys(rows);
  if (charProperties.length === 0) {
    return null;
  }

  return (
    <div className={`${className} detail-spec-table`}>
      <table className="detail-spec-table__table">
        <tbody>
          {charProperties.map((propName, index) => (<tr key={index}>
              <td>{propName}</td>
              <td>{rows[propName]}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};
