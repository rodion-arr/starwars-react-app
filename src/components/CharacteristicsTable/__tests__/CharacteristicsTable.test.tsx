import {render} from "@testing-library/react";
import {CharacteristicsTable} from "../CharacteristicsTable";

describe('CharacteristicsTable', () => {
  it('should render correctly', function () {
    const {container} = render(<CharacteristicsTable rows={{
      'test': 'value',
      'test2': 'value2',
    }} />);

    expect(container).toMatchSnapshot();
  });

  it('should not render on empty rows', function () {
    const {container} = render(<CharacteristicsTable rows={{}} />);

    expect(container).toMatchSnapshot();
  });
});
