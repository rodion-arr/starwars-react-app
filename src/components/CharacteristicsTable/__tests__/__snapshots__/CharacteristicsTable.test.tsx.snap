// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacteristicsTable should not render on empty rows 1`] = `<div />`;

exports[`CharacteristicsTable should render correctly 1`] = `
<div>
  <div
    class=" detail-spec-table"
  >
    <table
      class="detail-spec-table__table"
    >
      <tbody>
        <tr>
          <td>
            test
          </td>
          <td>
            value
          </td>
        </tr>
        <tr>
          <td>
            test2
          </td>
          <td>
            value2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
